### ISSUE TITLE
Report bad collection creation

### ISSUE THREAD

**Issue by orsinium** [wemake-services/wemake-python-styleguide#1913]
# Rule request

## Thesis

```python
# bad
lst = []
lst.append(1)
lst.append(2)

# good
lst = [1, 2]

# bad
st = set()
st.add(1)

# good
st = {1}

# bad
dc = {}
dc[1] = 2

# good
dc = {1: 2}
```

## Reasoning

Simpler, shorter, faster

**Comment by AlwxSin** [comment]
Here is a common case in my practice.
```python
def f():
  lst = []
  if some_cond:
    lst.append(1)
  return lst
```
Does linter able to detect these cases?

**Comment by sobolevn** [comment]
This will go into `typed-linter` at some point. It is impossible to reason about with raw ast.

**Comment by orsinium** [comment]
> This will go into typed-linter at some point. It is impossible to reason about with raw at.

Even if it is on the neighbored lines? My proposal is to ignore cases when there is anything in between.

**Comment by orsinium** [comment]
> Does linter able to detect these cases?

Sure thing, the task is only about explicit unconditional `list.append`, when it is right next to the variable initialization. I was lucky to face it a few times in the new project I work on :)
